src/unix/lwt_bytes.ml

```diff
@@ -240,7 +240,7 @@ external stub_mincore : t -> int -> int -> bool array -> unit = "lwt_unix_mincor
 let mincore buffer offset states =
   if (offset mod page_size <> 0
       || offset < 0
-      || offset > length buffer - (Array.length states * page_size)) then
+      || offset > length buffer) then
```

Comment:
I think we want to make sure that the offset is far back enough in the buffer, that for a flag array for N pages, the first N-1 pages are fully in memory, and there is at least one byte for the last page.

So, we want there to be at least `(Array.length states - 1) * page_size + 1` bytes in the buffer starting at `offset`. So, we need something like `length buffer - offset >= (Array.length states - 1) * page_size + 1`. The only thing to watch out for is that `Array.length states - 1` might be negative, if the array is empty. It seems like the check will still give us the right behavior, though, but we should probably leave a comment about that in the code.

Reply:
Ok, 
so if we want that there is at least `(Array.length states - 1) * page_size + 1` bytes in the buffer starting at `offset`, the condition that throws an exception for bad parameter is:

```
length buffer - offset < (Array.length states - 1) * page_size + 1
```
If I go through several cases like this:

* buff_len = 6 offset = 0 n_states = 1
```ocaml
6 - 0 < (1 - 1) * page_size + 1 (* -> false *)
```

* buff_len = page_size + 6 offset = 0 n_states = 1
```ocaml
page_size + 6 - 0 < (1 - 1) * page_size + 1 (* -> false *)
```

* buff_len = page_size + 6 offset = page_size n_states = 1

```ocaml
page_size + 6 - page_size < (1 - 1) * page_size + 1 (* -> false *)
```

* buff_len = page_size + 6 offset = page_size n_states = 2
```ocaml
page_size + 6 - page_size < (2 - 1) * page_size + 1 (*-> true (throw exception)*)
```

Do you want that I do all those tests (I have already almost done them) or do you see another case that are not covered?

Reply:
I'd say to set the offset always to page_size, then do:

1. buffer length = page_size * 2, n_states = 1 (should be ok)

2. buffer length = page_size * 2, n_states = 2 (should fail)

3. buffer length = page_size * 2 + 1, n_states = 2 (should be ok)

I think that exercises the main corner cases completely and exactly, though something may be missing.

We should also test what happens when n_states = 0.